app/models/user/reputation_tokens/exercise_contribution_token.rb

```diff
@@ -7,6 +7,7 @@ class User::ReputationTokens::ExerciseContributionToken < User::ReputationToken
   before_validation on: :create do
     self.track = contributorship.exercise.track unless track
     self.exercise = contributorship.exercise unless exercise
+    self.earned_on = self.exercise.updated_at unless earned_on
```

Comment:
1. Not `created_at`?
2. Why do you need `self.`?

Reply:
1. Updated
2. I don't. Removed.

At some point we should probably improve this a bit more to have the earned on data be set to the date at which the new contributor was synced.